smile_id_core/web_api.py

```diff
@@ -52,6 +55,67 @@ def validate_id_params(cls, data, service_request=False) -> dict:
 
         return IdValidation.validate(data, verification_schema)
 
+    def verify_document_async(
```

Comment:
Would propose extracting common code to a separate private method, something like this:
```python
class WebApiTest(ApiBase):
    ...
    def _prepare_document_verification_payload(
        country: str,
        id_type: str,
        id_number: str,
        first_name: str = None,
        last_name: str = None,
        dob: Union[str, date] = None,
        job_id: str = None,
        user_id: str = None,
    ):
    try:
        dob = dob.isoformat()
    except AttributeError:
        pass
    signature, timestamp = self._get_signature()
    ...
    return payload

    def verify_document(self, **data):
        payload = self._prepare_document_verification_payload(**data)
        response = self._make_request("POST", self.Urls.VERIFY_DOCUMENT, payload)
        ...
    
    def verify_document_async(self, **data):
        payload = self._prepare_document_verification_payload(**data)
        return self._make_request_async("POST", self.Urls.VERIFY_DOCUMENT_ASYNC, payload)
```
That way - less duplication and only one place where payload for the request is generated